tests/test_config_util.py

```diff
@@ -0,0 +1,66 @@
+# coding: utf-8
+
+
+__all__ = ["ConfigUtilTests"]
+
+import unittest
+
+from columnflow.util import maybe_import
+from columnflow.config_util import get_events_from_categories
+
+# from columnflow.analysis_templates.cms_minimal.__cf_module_name__.config.analysis__cf_short_name_lc__.py import cfg
+
+import order as od
+
+np = maybe_import("numpy")
+ak = maybe_import("awkward")
+dak = maybe_import("dask_awkward")
+coffea = maybe_import("coffea")
+
+
+class ConfigUtilTests(unittest.TestCase):
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        self.config_inst = cfg = od.Category("config", 1)
+
+        cfg.add_category("main_1", id=1)
+        main_2 = cfg.add_category("main_2", id=2)
+        main_2.add_category("leaf_21", id=21)
+        leaf_22 = main_2.add_category("leaf_22", id=22)
+        leaf_22.add_category("leaf_221", id=221)
+
+        # awkward array with category_ids (only leaf ids)
+        self.events = ak.Array({
+            "category_ids": [[1], [21, 221], [21], [221]],
+            "dummy_field": [1, 2, 3, 4],
+        })
+
+    def test_get_events_from_categories(self):
```

Comment:
You could probably split this test into each of the different blocks (i.e. smaller unit tests), but this is optional imho